infra/templates/az-isolated-service-single-region/tests/integration/integration_test.go

```diff
@@ -62,6 +66,42 @@ func httpGetRespondsWith200(goTest *testing.T, output infratests.TerraformOutput
 	}
 }
 
+// Verifies that the Key Vault instance deployed is properly isolated within the VNET
+func verifyVnetIntegrationForKeyVault(goTest *testing.T, output infratests.TerraformOutput) {
+	vaultResourceGroup := output["app_dev_resource_group"].(string)
+	vaultName := output["keyvault_name"].(string)
+	keyVaultACLs := azure.GetKeyVaultNetworkAcls(goTest, adminSubscription, vaultResourceGroup, vaultName)
+	subnetIDs := azure.GetSubnetsInVnet(goTest, adminSubscription, aseResourceGroup, os.Getenv("TF_VAR_ase_vnet_name"))
```

Comment:
as discussed, would be great to prefix the function name with the scoped resource type. For example `GetSubnetsInVnet` -> `VnetSubnetsList`,  `getVirtualNetworkClient` -> `vnetClient`